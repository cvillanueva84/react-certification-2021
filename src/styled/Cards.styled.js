import styled from 'styled-components';

export const StyledCardsGeneralContainer = styled.div`
  margin-top: 6rem;
  display: flex;
  height: 100%;
  width: 100%;
  justify-content: space-between;
  flex-flow: row wrap;
`;

export const StyledCardsContainer = styled.div`
  display: flex;
  height: 100%;
  width: 100%;
  justify-content: space-around;
  flex-flow: row wrap;
`;

export const StyledWelcomeHeader = styled.h1`
  width: 100%;
  font-weight: 200;
  text-align: center;
  font-size: 3.5rem;
`;

export const StyledCards = styled.div`
  margin: 10px;
  height: 345px;
  width: 345px;
  border-radius: 5px;
  box-shadow: 0 2px 0 lightgray;
  border: solid 1px lightgray;

  img {
    width: 100%;
    height: 40%;
    object-fit: cover;
    border-radius: 5px 5px 0 0;
  }

  h2 {
    margin: 0;
    font-weight: 400;
    font-size: 1.2rem;
  }

  p {
    font-size: small;
    color: gray;
  }

  div {
    margin: 1rem;
  }
`;
