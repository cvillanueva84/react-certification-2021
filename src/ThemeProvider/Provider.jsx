import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { AppContext } from '../context/appContext';
import { GlobalStyles } from './GlobalStyles.styled';
import { darkTheme, lightTheme } from './values';

const Provider = ({ children }) => {
  const { state } = useContext(AppContext);

  return (
    <ThemeProvider theme={state.theme ? lightTheme : darkTheme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>
  );
};

export default Provider;
