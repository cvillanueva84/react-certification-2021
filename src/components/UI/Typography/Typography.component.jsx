import React from 'react';
import { TypographyStyled } from './Typography.styled';

const Typography = (props) => {
  return <TypographyStyled {...props}>{props.children}</TypographyStyled>;
};

export default Typography;
