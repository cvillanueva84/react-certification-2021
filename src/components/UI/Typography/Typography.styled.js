import React from 'react';
import styled from 'styled-components';

const Header = ({ tagName = 'div', ...otherProps }) => {
  const Tag = tagName;

  return <Tag {...otherProps} />;
};

export const TypographyStyled = styled(Header)`
  color: ${(props) => props.theme.color};

  ${({ center }) => center && `text-align: center;`}

  ${(props) =>
    props.tiny &&
    `font-size: 12px;
      color: ${props.theme.textTiny};`}
  
    ${({ weight }) => {
    switch (weight) {
      case '100':
        return `font-weight: 100;`;
      case '200':
        return `font-weight: 200;`;
      case '300':
        return `font-weight: 300;`;
      case '400':
        return `font-weight: 400;`;
      case '500':
        return `font-weight: 500;`;
      case '600':
        return `font-weight: 600;`;
      case '700':
        return `font-weight: 700;`;
      case '800':
        return `font-weight: 800;`;
      case '900':
        return `font-weight: 900;`;
      default:
        break;
    }
  }}
`;
