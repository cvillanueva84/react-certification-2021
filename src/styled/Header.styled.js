import styled from 'styled-components';

const StyledHeader = styled.header`
  color: white;
  display: flex;
  height: 3rem;
  width: 100%;
  background-color: #1c5476;
  justify-content: space-between;
  align-items: center;
`;

export default StyledHeader;
