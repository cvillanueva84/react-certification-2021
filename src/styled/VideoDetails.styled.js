import styled from 'styled-components';

export const StyledVideoDetails = styled.div`
  display: flex;
  position: relative;
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  color: black;
  flex-direction: row;
  height: 100%;
  width: 100%;
  margin-top: 54px;
  overflow: scroll;

  .video-container {
    width: 70%;
  }

  iframe {
    height: 500px;
    width: 100%;
    border-width: 2px;
  }

  h2 {
    font-size: 1.5rem;
    font-weight: 400;
    line-height: 1.334;
    letter-spacing: 0em;
  }

  p {
    color: rgba(0, 0, 0, 0.54);
    font-size: 0.875rem;
    font-weight: 400;
    line-height: 1.43;
    letter-spacing: 0.01071em;
  }
`;

export const StyledRecommendedBar = styled.div`
  width: 30%;
  flex-direction: column;

  .recommended-box {
    display: flex;
    flex-direction: row;
    height: 100px;
    overflow: hidden;
    align-items: center;
  }

  img {
    width: 120px;
    height: 90px;
    object-fit: cover;
  }
  p {
    color: black;
    font-size: 1rem;
    font-weight: 400;
    line-height: 1.5;
    letter-spacing: 0.00938em;
  }
`;
