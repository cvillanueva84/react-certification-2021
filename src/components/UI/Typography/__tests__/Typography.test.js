import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import Typography from '../index';
import { GlobalStyles } from '../../../../GlobalStyles.styled';
import { darkTheme, lightTheme } from '../../../../ThemeProvider/values';

describe('Testing Typography component', () => {
  const theme = true;
  let tree;

  beforeEach(() => {
    tree = render(
      <ThemeProvider theme={theme ? lightTheme : darkTheme}>
        <GlobalStyles />
        <Typography center="true">Some</Typography>
        <Typography header="true">Some</Typography>
        <Typography tiny="true">Some</Typography>

        <Typography weight="100">Some</Typography>
        <Typography weight="200">Some</Typography>
        <Typography weight="300">Some</Typography>
        <Typography weight="400">Some</Typography>
        <Typography weight="500">Some</Typography>
        <Typography weight="600">Some</Typography>
        <Typography weight="700">Some</Typography>
        <Typography weight="800">Some</Typography>
        <Typography weight="900">Some</Typography>
      </ThemeProvider>
    );
  });

  it('should to take snapshop', () => {
    expect(tree).toMatchSnapshot();
  });
});
