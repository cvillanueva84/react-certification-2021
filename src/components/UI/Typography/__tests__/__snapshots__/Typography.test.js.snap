// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Typography component should to take snapshop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        center="true"
        class="sc-bdnxRM bmgBwX"
      >
        Some
      </div>
      <div
        class="sc-bdnxRM eukGYp"
        header="true"
      >
        Some
      </div>
      <div
        class="sc-bdnxRM dqhziO"
        tiny="true"
      >
        Some
      </div>
      <div
        class="sc-bdnxRM hdAjci"
        weight="100"
      >
        Some
      </div>
      <div
        class="sc-bdnxRM kouqyJ"
        weight="200"
      >
        Some
      </div>
      <div
        class="sc-bdnxRM bcQeOI"
        weight="300"
      >
        Some
      </div>
      <div
        class="sc-bdnxRM bbKXvX"
        weight="400"
      >
        Some
      </div>
      <div
        class="sc-bdnxRM cgbnZy"
        weight="500"
      >
        Some
      </div>
      <div
        class="sc-bdnxRM fqVvvZ"
        weight="600"
      >
        Some
      </div>
      <div
        class="sc-bdnxRM huMTkU"
        weight="700"
      >
        Some
      </div>
      <div
        class="sc-bdnxRM kfWCJT"
        weight="800"
      >
        Some
      </div>
      <div
        class="sc-bdnxRM lkmTnu"
        weight="900"
      >
        Some
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      center="true"
      class="sc-bdnxRM bmgBwX"
    >
      Some
    </div>
    <div
      class="sc-bdnxRM eukGYp"
      header="true"
    >
      Some
    </div>
    <div
      class="sc-bdnxRM dqhziO"
      tiny="true"
    >
      Some
    </div>
    <div
      class="sc-bdnxRM hdAjci"
      weight="100"
    >
      Some
    </div>
    <div
      class="sc-bdnxRM kouqyJ"
      weight="200"
    >
      Some
    </div>
    <div
      class="sc-bdnxRM bcQeOI"
      weight="300"
    >
      Some
    </div>
    <div
      class="sc-bdnxRM bbKXvX"
      weight="400"
    >
      Some
    </div>
    <div
      class="sc-bdnxRM cgbnZy"
      weight="500"
    >
      Some
    </div>
    <div
      class="sc-bdnxRM fqVvvZ"
      weight="600"
    >
      Some
    </div>
    <div
      class="sc-bdnxRM huMTkU"
      weight="700"
    >
      Some
    </div>
    <div
      class="sc-bdnxRM kfWCJT"
      weight="800"
    >
      Some
    </div>
    <div
      class="sc-bdnxRM lkmTnu"
      weight="900"
    >
      Some
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
